rust/datafusion/src/physical_plan/hash_join.rs

```diff
@@ -311,6 +319,7 @@ fn update_hash(
     hash: &mut JoinHashMap,
     offset: usize,
     random_state: &RandomState,
+    hashes_buffer: &mut Vec<u64>,
```

Comment:
This is effectively allowing `hashes_buffer ` to be reused, right?

It may eventually make sense to make some struct that holds all the relevant state (`on`, `random_state`, `hash_buf`, etc).

Reply:
Indeed, this change is for reusing the allocated `Vec`.

Yes, makes sense to group them in a struct. There are some opportunities in other functions "build_join_indexes"  "build_batch", etc. for this as well. Not sure if it makes sense they all receive the same struct, or maybe all of them a subset of the most commonly needed parts :thinking:

Reply:
🤔  definitely not for this PR